packages/openmrs/src/Utils.js

```diff
@@ -18,20 +17,45 @@ export const prepareNextState = (state, response, callback = s => s) => {
 export async function fetchAndLog(method, url, opts) {
   try {
     const response = await commonRequest(method, url, opts);
-    logResponse(response);
+    const { statusCode, duration } = response;
+
+    const urlWithQuery = Object.keys(opts.query || {}).length
+      ? `${url}?${new URLSearchParams(opts.query).toString()}`
+      : url;
+
+    const message = `${method} ${urlWithQuery} - ${statusCode} in ${duration}ms`;
+    if (response instanceof Error) {
+      console.error(message);
+    } else {
+      console.log(message);
+    }
+    response.url = urlWithQuery;
     return response;
   } catch (err) {
     return err;
   }
 }
 
+// TODO get() should pass autopage true and options
+// TODO http.*() functions should all pass autopage false (at least for now)
+// TODO this currently returns { result, response }. Is that right? It'll break everything!!
```

Comment:
I think this is correct, yes, and so now we have to go and fix everything.

- we use `result` because response bodies may be aggregated, if using pagination.
- `response` right now might be an object or array, if paginate is enabled

Would it be cleaner to have `request()` and `paginatedGet()` as two distinct APIs, with two distinct return types?

That would mean that the main rest `get()` explictly calls paginated get, and basically every other API just calls get. The http helpers will not autopage for you. This is a nice clear seperation of concerns and makes the behaviour and implementation way simpler.

Ok, I think I like that. But then I am very tired.